File: src/components/cameraview_py/main/GUI.py
from PyQt4 import QtCore
from PyQt4 import QtGui


class GUI(QtGui.QWidget):
    def __init__(self, sensor, parent=None):

        self.sensor = sensor

        QtGui.QWidget.__init__(self, parent)

        self.InitUI();
        
    def InitUI(self):

        self.labelImage = QtGui.QLabel();
                            
        mainLayout = QtGui.QGridLayout()
        
        mainLayout.addWidget(self.labelImage, 0, 0);

        self.setLayout(mainLayout)
        
        QtCore.QObject.connect(self, QtCore.SIGNAL("aa"), self.update)

        
        #self.setImage()
        
    def updateImage(self):
        img = self.sensor.getImage()
        image = QtGui.QImage(img.data, img.shape[1], img.shape[0], img.shape[1]*img.shape[2], QtGui.QImage.Format_RGB888);
        self.labelImage.setPixmap(QtGui.QPixmap.fromImage(image))
    
    def update(self):
        self.updateImage()
        
    def updateGUI_recieved_signal(self):
        self.emit(QtCore.SIGNAL("aa"), "lol")

File: src/stable/components/basic_component_qt/gui/gui.cpp
#include "gui.h"
namespace basic_component_qt {
Gui::Gui(Shared* sm)
{

    this->sm = sm;

    //Creation and setup of Gui elements
    QGridLayout* mainLayout = new QGridLayout();
    labelImage = new QLabel();
    mainLayout->addWidget(labelImage, 0, 0);
    setLayout(mainLayout);
    setVisible(true);

    //We connect a signat to a slot to be able to do something from an incoming signal.
    connect(this, SIGNAL(signal_updateGUI()), this, SLOT(on_updateGUI_recieved()));

    show();

}


void Gui::updateThreadGUI()
{
    //emit the signal to update the gui
    emit signal_updateGUI();
}

void Gui::on_updateGUI_recieved()
{
    //Create and displays the image taken from the shared memory.

    //cv::Mat
    cv::Mat frame = this->sm->getImage();
    QImage imageQt = QImage((const unsigned char*)(frame.data),
                            frame.cols,
                            frame.rows,
                            frame.step,
                            QImage::Format_RGB888);
/*  
    //IplImage
    IplImage* frame = this->sm->getImage();
    QImage imageQt = QImage((const unsigned char*)(frame->imageData),
                            frame->width,
                            frame->height,
                            QImage::Format_RGB888);
*/
  
    labelImage->setPixmap(QPixmap::fromImage(imageQt));
}

}
